editor/plugins/script_text_editor.cpp

```diff
@@ -396,8 +396,215 @@ bool ScriptTextEditor::show_members_overview() {
+		inline_color_line = p_info[TextEdit::INFO_LINE];
+		inline_color_start = p_info[TextEdit::INFO_COLUMN];
+		inline_color_end = p_info[INFO_COLOR_END];
+
+		// Move popup above the line if it's too low.
+		float_t view_h = get_viewport_rect().size.y;
+		float_t pop_h = inline_color_popup->get_contents_minimum_size().y;
+		float_t pop_y = p_rect.get_end().y;
+		float_t pop_x = p_rect.position.x;
+		if (pop_y + pop_h > view_h) {
+			pop_y = p_rect.position.y - pop_h;
+		}
+		// Move popup to the right if it's too high.
+		if (pop_y < 0) {
+			pop_x = p_rect.get_end().x;
+		}
+
+		inline_color_popup->popup(Rect2(pop_x, pop_y, 0, 0));
+		code_editor->get_text_editor()->set_selection_mode(TextEdit::SelectionMode::SELECTION_MODE_NONE);
+	}
+}
+
+String ScriptTextEditor::_picker_color_stringify(const Color &p_color, int p_mode) {
+	String result;
+	String fname;
+	Vector<String> str_params;
+	// HTML string.
+	if (p_mode < 0) {
+		str_params.push_back("\"" + p_color.to_html() + "\"");
+	}
+	// RGBA int parameters (0-255).
+	else if (p_mode == ColorPicker::MODE_RGB) {
+		str_params = {
+			itos(p_color.get_r8()),
+			itos(p_color.get_g8()),
+			itos(p_color.get_b8()),
+			itos(p_color.get_a8())
+		};
+		fname = ".from_rgba8";
+	}
+	// HSL, HSV, and RAW Float parameters.
+	else {
+		PackedFloat64Array flt_params;
+		str_params.resize(4);
+		switch (p_mode) {
+			case ColorPicker::MODE_HSV: {
+				flt_params = { p_color.get_h(), p_color.get_s(), p_color.get_v(), p_color.a };
+				fname = ".from_hsv";
+			} break;
+			case ColorPicker::MODE_OKHSL: {
+				flt_params = { p_color.get_ok_hsl_h(), p_color.get_ok_hsl_s(), p_color.get_ok_hsl_l(), p_color.a };
+				fname = ".from_ok_hsl";
+			} break;
+			case ColorPicker::MODE_RAW: {
+				flt_params = { p_color.r, p_color.g, p_color.b, p_color.a };
+				fname = "";
+			} break;
+		}
+		for (int ind = 0; ind < flt_params.size(); ind++) {
+			str_params.set(ind, String::num(flt_params[ind], 3));
+		}
+	}
+	result = "Color" + fname + "(" + String(", ").join(str_params) + ")";
+	return result;
+}
+
+void ScriptTextEditor::_picker_color_changed(const Color &p_color) {
+	if (inline_color_line < 0) {
+		return;
+	}
+	int mode = inline_color_picker->get_color_mode();
+	if (inline_color_picker->is_using_hex_string()) {
+		mode = -1;
+	}
+	String result = _picker_color_stringify(p_color, mode);
+	code_editor->get_text_editor()->begin_complex_operation();
+	code_editor->get_text_editor()->remove_text(inline_color_line, inline_color_start, inline_color_line, inline_color_end + 1);
+	inline_color_end = inline_color_start + result.size() - 2;
+	code_editor->get_text_editor()->insert_text(result, inline_color_line, inline_color_start);
+	code_editor->get_text_editor()->end_complex_operation();
+}
+
+void ScriptTextEditor::_picker_opening() {
+	previous_tooltip_enabled = code_editor->get_text_editor()->is_symbol_tooltip_on_hover_enabled();
+	code_editor->get_text_editor()->set_symbol_tooltip_on_hover_enabled(false);
+}
```

Comment:
If this is called when the picker is already open, it gets called before `_picker_closed` and causes the symbol tooltip on hover to stay off.